ws-server/src/server.rs

```diff
@@ -209,16 +211,19 @@ async fn background_task(
 	conn_id: ConnectionId,
 	methods: Methods,
 	max_request_body_size: u32,
-	stop_monitor: StopMonitor,
+	stop_server: StopMonitor,
 ) -> Result<(), Error> {
 	// And we can finally transition to a websocket background_task.
 	let (mut sender, mut receiver) = server.into_builder().finish();
 	let (tx, mut rx) = mpsc::unbounded::<String>();
+	let stop_conn = Arc::new(AtomicBool::new(false));
+
+	let stop_server2 = stop_server.clone();
+	let stop_conn2 = stop_conn.clone();
```

Comment:
we need this flag to stop the inner task that holds the `receiver` of the channel when the connection is dropped by the client.